### PR TITLE
Added Vulcan functionality to Observation and modified related classes…

### PR DESCRIPTION
**Pull request by that-guy-is-it**
… accordingly. Also added relevant tests for the expanded search, which now supports CSV values.

### CODE REVIEW

File: patient-generated-data-tests/src/main/java/gov/va/api/health/patientgenerateddata/tests/SystemDefinitions.java [hunk 1]
```diff
@@ -14,6 +14,8 @@
   private static Ids ids() {
     return Ids.builder()
         .observation("fc691a7f-a0f3-47b4-9d00-2786d055e8ba")
+        .observationList(
+            "fc691a7f-a0f3-47b4-9d00-2786d055e8ba," + "0b9d2e37-f84d-4f9e-9ba3-995772f368d3")
```
Comment (apeirats-va) [inline patient-generated-data-tests/src/main/java/gov/va/api/health/patientgenerateddata/tests/SystemDefinitions.java:18]: Could make this a `List.of` and then `.join` it later

File: patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationController.java [hunk 2]
```diff
@@ -65,6 +72,15 @@ public static ObservationEntity toEntity(Observation observation) {
     return populate(observation, ObservationEntity.builder().id(observation.id()).build());
   }
 
+  private VulcanConfiguration<ObservationEntity> configuration() {
+    return VulcanConfiguration.forEntity(ObservationEntity.class)
+        .paging(linkProperties.pagingConfiguration("Observation", ObservationEntity.naturalOrder()))
+        .mappings(Mappings.forEntity(ObservationEntity.class).csvList("id", "id").get())
```
Comment (apeirats-va) [inline patient-generated-data/src/main/java/gov/va/api/health/patientgenerateddata/observation/ObservationController.java:78]: The search param should be `_id` I think.

File: patient-generated-data/src/test/java/gov/va/api/health/patientgenerateddata/observation/ObservationControllerTest.java [hunk 3]
```diff
@@ -14,15 +15,54 @@
...
+    request.setRequestURI(u.getPath());
+    request.setRemoteHost(u.getHost());
+    request.setProtocol(u.getScheme());
+    request.setServerPort(u.getPort());
+    u.getQueryParams()
+        .entrySet()
+        .forEach(e -> request.addParameter(e.getKey(), e.getValue().toArray(new String[0])));
+    return request;
+  }
+
```
Comment (apeirats-va) [inline patient-generated-data/src/test/java/gov/va/api/health/patientgenerateddata/observation/ObservationControllerTest.java:56]: We now have a copy of the `requestFromUri` method on 3 separate classes. Maybe it's time to generalize it elsewhere

File: patient-generated-data-tests/src/main/java/gov/va/api/health/patientgenerateddata/tests/SystemDefinitions.java [hunk 4]
```diff
@@ -158,6 +161,8 @@ static SystemDefinition systemDefinition() {
   static final class Ids {
     @NonNull String observation;
 
+    @NonNull List<String> observationList;
```
Comment (jblefkove-va) [inline patient-generated-data-tests/src/main/java/gov/va/api/health/patientgenerateddata/tests/SystemDefinitions.java:164]: this is a list, but `questionnaireList` and `questionnaireResponseList` are strings

File: patient-generated-data/src/test/java/gov/va/api/health/patientgenerateddata/MockRequests.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package gov.va.api.health.patientgenerateddata;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.util.UriComponentsBuilder;
+
+public class MockRequests {
+
+  public static MockHttpServletRequest requestFromUri(String uri) {
```
Comment (jblefkove-va) [inline patient-generated-data/src/test/java/gov/va/api/health/patientgenerateddata/MockRequests.java:8]: should `CompositeMappingJpaTest` and `TokenListMappingJpaTest` use this as well?

File: patient-generated-data-tests/src/main/java/gov/va/api/health/patientgenerateddata/tests/SystemDefinitions.java (hunk 1 above)
Comment (jblefkove-va) [reply patient-generated-data-tests/src/main/java/gov/va/api/health/patientgenerateddata/tests/SystemDefinitions.java]: Just saw this comment... `List<String>` is probably better, but I would prefer this also be consistent with `questionnaireList` and `questionnaireResponseList`

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**